db/verifications.go

```diff
@@ -46,17 +54,17 @@ func (ms *MongoStorage) UserVerificationCode(user *User, t CodeType) (string, er
 	verification := &UserVerification{}
 	if err := result.Decode(verification); err != nil {
 		if err == mongo.ErrNoDocuments {
-			return "", ErrNotFound
+			return nil, ErrNotFound
 		}
-		return "", err
+		return nil, err
 	}
-	return verification.Code, nil
+	return verification, nil
 }
 
 // SetVerificationCode method sets the verification code for the user provided.
 // If the user already has a verification code, it updates it. If an error
 // occurs, it returns the error.
-func (ms *MongoStorage) SetVerificationCode(user *User, code string, t CodeType) error {
+func (ms *MongoStorage) SetVerificationCode(user *User, code string, t CodeType, exp time.Time) error {
```

Comment:
@lucasmenendez maybe a bit late for this comment, but is there a reason that the function waits for `*User` and not a userID ?
Same question applies for `UserVerifiactionCode` function

Reply:
I think that receives the user struct is more flexible, it does not depends of a single information about the user but also allows to set an `*User` instance with just the `userID`.